ui/src/views/mint/github-step/github-repository-selection.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  Button,
+  Card,
+  Combobox,
+  ComboboxItem,
+  Flex,
+  Grid,
+  Icon,
+  IconButton,
+} from '@/components';
+import { Input } from '@/components/core/input';
+import { Separator } from '@/components/core/separator.styles';
+import React, { forwardRef, useRef, useState } from 'react';
+import { Mint } from '../mint.context';
+
+//TODO remove once it's integrated with GH login
+const repos = [
+  'DyDx',
+  'Testing',
+  'Hello World',
+  'Portofolio',
+  'NFA',
+  'NFT',
+  'NFTs',
+];
+
+//TODO remove once it's integrated with GH login
+const users: ComboboxItem[] = [
+  { label: 'DyDx', value: 'DyDx', icon: 'github' },
+  { label: 'Testing', value: 'Testing', icon: 'github' },
+  { label: 'Hello World', value: 'Hello World', icon: 'github' },
+  { label: 'Portofolio', value: 'Portofolio', icon: 'github' },
+  { label: 'NFA', value: 'NFA', icon: 'github' },
+  { label: 'NFT', value: 'NFT', icon: 'github' },
+  { label: 'NFTs', value: 'NFTs', icon: 'github' },
+];
+
+type RepoRowProps = {
+  repo: string;
+  button: React.ReactNode;
+} & React.ComponentProps<typeof Flex>;
+
+export const RepoRow = forwardRef<HTMLDivElement, RepoRowProps>(
+  ({ repo, button, ...props }, ref) => (
+    <Flex
+      {...props}
+      ref={ref}
+      css={{ justifyContent: 'space-between', my: '$4', ...props.css }}
+    >
+      <Flex css={{ alignItems: 'center' }}>
+        <Icon name="github" css={{ fontSize: '$2xl', mr: '$2' }} />
+        <span>{repo}</span>
+      </Flex>
+      {button}
+    </Flex>
+  )
+);
+
+export const GithubRepositoryConnection: React.FC = () => {
+  const [searchValue, setSearchValue] = useState('');
+  const [selectedUser, setSelectedUser] = useState<ComboboxItem | undefined>();
+  const { setGithubStep, setRepositoryName } = Mint.useContext();
+
+  const timeOutRef = useRef<NodeJS.Timeout>();
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.stopPropagation();
+    timeOutRef.current && clearTimeout(timeOutRef.current);
+    timeOutRef.current = setTimeout(() => {
+      setSearchValue(event.target.value);
+    }, 500);
+  };
+
+  const handlePrevStepClick = () => {
+    setGithubStep(1);
+  };
+
+  const handleSelectRepo = (repo: string) => {
+    setRepositoryName(repo);
+    setGithubStep(3);
+  };
+
+  return (
+    <Card.Container css={{ maxWidth: '$107h', height: '$95h', pb: '$0h' }}>

```

Comment:
The same from the previous component, we may split this in different component (if the component grow up too much) to keep handlers separated and make it more readable.